modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/complex/JsonSelectFunction.java

```diff
@@ -52,11 +52,9 @@ public Object evaluate(Object object) {
         String path = (String) this.params.get(0).evaluate(object);
         if (object instanceof MongoCollectionFeature) {
             // in the case of a mongo collection we need to append the parent path
-            if (!MongoComplexUtilities.useLegacyPaths()) {
                 String parentPath = ((MongoCollectionFeature) object).getCollectionPath();
```

Comment:
Looks the spacing was not fixed, let's see if the build complains.

Reply:
fixed thanks